code/authenticators/SAMLSecurityExtension.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+class SAMLSecurityExtension extends Extension {
+
+	/**
+	 * Will redirect the user directly to the ADFS if:
+	 *
+	 * 1) the 'SAMLAuthenticator' is the default authenticator
+	 * 2) there isn't a GET param showloginform set to 1
+	 * 3) the member is currently logged in
+	 * 4) there are form messages (errors or notices)
+	 *
+	 * @return void
+	 */
+	public function onBeforeSecurityLogin() {
+
+		if(Authenticator::get_default_authenticator() != 'SAMLAuthenticator') {
+			return;
+		}
+
+		// by going to the URL Security/login?showloginform=1 we bypass the auto sign on
+		if(isset($_GET['showloginform']) && $_GET['showloginform'] == 1) {
```

Comment:
That is weird, I used to not get a proper request object back (`NullHTTPRequest`), now I do. I'll change that.